test/factories/sample.js

```diff
@@ -0,0 +1,5 @@
+import uuid from 'node-uuid';
+
+export default Factory =>
+  Factory.define('sample')
+    .attr('id', () => uuid.v4());
```

Comment:
`uuid.v4()` is just a method on the `node-uuid` module (see line 1). It's an NPM package that generates UUIDs for you. For the factories (which use Rosie, another NPM package) you can just any values you want in. For example, you might want a plain string, a `true` value, or to generate a random string using another package.

In terms of data types in the database, these are predominantly defined by Knex (which is the query builder and handles things like migrations and adding seed data) and generally mirror types available in Postgres databases (SQL+). I'll add a link to the Knex documentation somewhere below.

Also, there are model data types to consider. Whilst these loosely mirror the same types available in Knex (and by extension the database itself) there are some notable differences. I'll add a link to the Bookshelf docs too.